src/modules/rename-state.ts

```diff
@@ -0,0 +1,56 @@
+import { codeToAst } from "../parsing";
+import { selectedText, allText, selectedTextStart, selectedTextEnd, activeFileName, activeEditor, showInputBox } from "../editor";
+import traverse from "@babel/traverse";
+import { transformFromAst } from "@babel/core";
+import { window, Position, Range } from "vscode";
+import { persistFileSystemChanges, replaceTextInFile } from "../file-system";
+import { capitalizeFirstLetter } from "../utils";
+import * as t from "@babel/types";
+import { findPathInContext } from "../ast-helpers";
+
+function isPathOnLines(path, start, end) {
+    if (!path.node) return false;
+    const pathStart = path.node.loc.start;
+    const pathEnd = path.node.loc.end;
+    return (
+        (pathStart.line === start.line && pathStart.column === start.character) &&
+        (pathEnd.line === end.line && pathEnd.column === end.character))
+
+}
+
+export function isStateVariable(text) {
+    try {
+        const allAST = codeToAst(allText());
+        const containerPath = findPathInContext(allAST, text);
+        const variableDeclarationParent = containerPath.scope.bindings[containerPath.node.name].path.parentPath;
+        return variableDeclarationParent && variableDeclarationParent.node.declarations[0].init.callee.name === 'useState'
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['magic numbers' are not allowed: 0](https://app.codacy.com/gh/wix/vscode-glean/pullRequest?prid=5535229)